Swift/Function.swift

```diff
@@ -39,8 +39,12 @@ public final class Function {
     ///
     /// - Parameter expression: The expression.
     /// - Returns: The COUNT(expr) function.
-    public static func count(_ expression: ExpressionProtocol) -> ExpressionProtocol {
-        return QueryExpression(CBLQueryFunction.count(expression.toImpl()))
+    public static func count(_ expression: ExpressionProtocol? = nil) -> ExpressionProtocol {
+        if let expression = expression {
```

Comment:
I'm not sure what idiomatic Obj-C looks like so ignore this if it is irrelevant.
I would be inclined to have the value of the `let` be either expression.toImpl() or expression.property('*) and to have a single return statement.